addOns/ascanrulesAlpha/src/main/java/org/zaproxy/zap/extension/ascanrulesAlpha/ForbiddenBypassScanRule.java

```diff
@@ -0,0 +1,193 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.ascanrulesAlpha;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.LinkedList;
+import java.util.List;
+import org.apache.log4j.Logger;
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.core.scanner.AbstractAppPlugin;
+import org.parosproxy.paros.core.scanner.Alert;
+import org.parosproxy.paros.core.scanner.Category;
+import org.parosproxy.paros.network.HttpMessage;
+import org.parosproxy.paros.network.HttpStatusCode;
+
+/**
+ * Active scan rule for Reverse proxy testing.
+ *
+ * @author Aman Rawat (@theamanrawat)
+ */
+public class ForbiddenBypassScanRule extends AbstractAppPlugin {
+
+    private static Logger log = Logger.getLogger(ForbiddenBypassScanRule.class);
+
+    @Override
+    public int getId() {
+        return 40038;
+    }
+
+    @Override
+    public String getName() {
+        return Constant.messages.getString("ascanalpha.forbiddenBypass.name");
+    }
+
+    @Override
+    public String getDescription() {
+        return Constant.messages.getString("ascanalpha.forbiddenBypass.desc");
+    }
+
+    @Override
+    public void scan() {
+        HttpMessage message = getNewMsg();
+        try {
+            sendAndReceive(message);
+            if (message.getResponseHeader().getStatusCode() == HttpStatusCode.FORBIDDEN) {
+                String FULL_URL = message.getRequestHeader().getURI().toString();
+                URI uri = new URI(FULL_URL);
+                String HOST = uri.getAuthority();
+                String PATH = uri.getPath();
+                String SCHEMA = uri.getScheme();
+                List<String> upayloads =
+                        new LinkedList<String>(
+                                Arrays.asList(
+                                        new String[] {
+                                            "/%2e" + PATH,
+                                            PATH + "/.",
+                                            "/." + PATH + "/./",
+                                            PATH + "%20/",
+                                            "/%20" + PATH + "%20/",
+                                            PATH + "..;",
+                                            PATH + "?",
+                                            PATH + "??",
+                                            "/" + PATH + "//",
+                                            PATH + "/",
+                                            PATH + "/.testus",
+                                            PATH + "../app.py"
+                                        }));
+                List<String> hpayloads =
```

Comment:
- `headerPayloads` 
 - This can be declared later. I'd suggest extract methods for each test (path and header).